Dockerfile

```diff
@@ -34,11 +34,6 @@ RUN --mount=type=cache,target=/go/pkg/mod \
 # Copy the sources
 COPY ./ ./
 
```

Comment:
Yeah, podman doesn't use this file. I was looking at Dockerfile because I originally trying to create a combined file, but I couldn't make the caching work.

I removed the additional go build step because afaict it's redundant even for docker. If you notice we build again with build arguments below and the same cache. This step is likely caching artifacts which we don't use. The 'real' build will create artifacts for everything we actually use, and nothing else. This step can only result in additional work and doesn't improve cache effectiveness.

...unless I've misunderstood some semantics of buildkit? In which case we can revert this.